### PR TITLE
Fixes #12129 Change sync status to use node on products base rabl tem…

### PR DESCRIPTION
**Pull request by johnpmitsch**
…plate

using attributes with sync status is causing rails 4 to error when calling to_json. This can be fixed by using a node.

Opening this up on master to ease the rails 4 transition.

### CODE REVIEW

**Comment by cpeters** [comment]
ACK

File: app/views/katello/api/v2/products/base.json.rabl [hunk 1]
```diff
@@ -7,13 +7,30 @@ extends 'katello/api/v2/common/org_reference'
...
 attributes :provider_id
 attributes :sync_plan_id
-attributes :sync_status
 attributes :sync_summary
 attributes :gpg_key_id
 attributes :redhat? => :redhat
 
 attributes :available_content => :available_content, :if => params[:include_available_content]
 
+node :sync_status do |product|
```
Comment (jlsherrill) [inline app/views/katello/api/v2/products/base.json.rabl]: does child not work?

Comment (johnpmitsch) [inline app/views/katello/api/v2/products/base.json.rabl]: it doesn't because its not a relation to a product

Comment (johnpmitsch) [inline app/views/katello/api/v2/products/base.json.rabl]: thats my guess of why at least, here is what I am seeing in the rendered json when I do child :sync_status http://fpaste.org/277161/44096471/

Comment (jlsherrill) [inline app/views/katello/api/v2/products/base.json.rabl]: oh right, that function isn't returning a presenter object, its just returning a hash.

File: app/views/katello/api/v2/products/base.json.rabl [hunk 2]
```diff
@@ -7,13 +7,30 @@ extends 'katello/api/v2/common/org_reference'
...
+    :raw_state => product.sync_status[:raw_state],
+    :start_time => product.sync_status[:start_time],
+    :finish_time => product.sync_status[:finish_time],
+    :duration => product.sync_status[:duration],
+    :display_size => product.sync_status[:display_size],
+    :size => product.sync_status[:size],
+    :is_running => product.sync_status[:is_running],
+    :error_details => product.sync_status[:error_details]
+  }
+end
```
Comment (jlsherrill) [inline app/views/katello/api/v2/products/base.json.rabl]: You might simply add a local variable:
```
status = product.sync_status 
```

right before the hash, would simplify the hash quite a bit

**Comment by johnpmitsch** [comment]
@jlsherrill updated

**Comment by cpeters** [comment]
I'm actually against this change. Now when I read the code I have to remember that status is actually referencing a hash on the product. Whereas the previous version I just read and parsed; don't have to remember the context change.

**Comment by jlsherrill** [comment]
So another reason for that change is that calling that function that many times results in many many more queries, as even though Product#last_repo_sync_task is cached, things like the lock lookup and repository list are actual queries.  

We could cache the results of that function, but that seems a tad less than ideal

**Comment by cpeters** [comment]
Gotcha; previous statement retracted.

**Comment by jlsherrill** [comment]
ACK

**Comment by johnpmitsch** [comment]
@jlsherrill I don't have merge permissions :sob: can you plz merge? :)

**Pull request merged**